src/api/java/com/minecolonies/coremod/entity/ai/util/StructureIterator.java

```diff
@@ -158,7 +157,7 @@ public boolean doesStructureBlockEqualWorldBlock()
             }
             else if ((structureBlock instanceof BlockStairs && structureBlockState.equals(worldBlockState))
                        || BlockUtils.isGrassOrDirt(structureBlock, worldBlock, structureBlockState, worldBlockState)
-                       || (worldBlock == ModBlocks.blockRack && BlockMinecoloniesRack.shouldBlockBeReplacedWithRack(structureBlock)))
+                       || (worldBlock instanceof IBlockMinecoloniesRack && IBlockMinecoloniesRack.shouldBlockBeReplacedWithRack(structureBlock)))
```

Comment:
Isn't that less efficient?

Reply:
Not really.
The difference is not notable.

It allows for easier extension in the future as well.

Reply:
afaik instanceof is pretty fast these days